### PR TITLE
CB-596 feat: pagination for database objects

### PR DESCRIPTION
**Pull request by Wroud**

### CODE REVIEW

File: webapp/packages/core-app/src/CoreSettingsService.ts [hunk 1]
```diff
@@ -13,8 +13,7 @@ const defaultSettings = {
   'app.logViewer.maxLogRecords': 1000,
   'app.logViewer.logBatchSize': 2000,
   'app.logViewer.maxFailedRequests': 3,
-  // temporary limit for all nodes children in app
-  'app.navigationTree.childrenLimit': 1000,
+  'app.navigationTree.childrenLimit': 100,
```
Comment (tfokina) [inline webapp/packages/core-app/src/CoreSettingsService.ts:16]: why did you decrease the limit?

File: webapp/packages/plugin-navigation-tree/src/NavigationTree/ElementsTree/NavTreeLimitFilter/elementsTreeLimitRenderer.tsx [hunk 2]
```diff
@@ -6,35 +6,42 @@
...
-`;
 
-const ManageableGroup: NavigationNodeRendererComponent = observer(function ManageableGroup() {
-  const navTreeResource = useService(NavTreeResource);
-  return styled(styles)(
-    <connection-group>
-      <Translate token="app_navigationTree_limited" limit={navTreeResource.childrenLimit} />
-    </connection-group>,
+  return (
+    <div className={s(styles, { loadMoreBox: true })}>
```
Comment (tfokina) [inline webapp/packages/plugin-navigation-tree/src/NavigationTree/ElementsTree/NavTreeLimitFilter/elementsTreeLimitRenderer.tsx]: add data-testid maybe?

**Review by tfokina — APPROVED** [review]

File: webapp/packages/core-sdk/src/Resource/ResourceAlias.ts [hunk 3]
```diff
@@ -15,25 +15,72 @@ export type ResourceAliasOptions = Readonly<Record<ResourceAliasOptionsKey, Reso
...
+
+  setParent(parent: ResourceAlias<TKey, any>): this {
+    const copy = new (this.constructor as any)(this.id, this.options, parent) as this;
+    return copy.setTarget(this.target);
   }
 
   isEqual(key: ResourceAlias<TKey, any>): boolean {
-    if (isResourceAlias(key)) {
-      return key.id === this.id && isObjectsEqual(this.options, key.options);
+    if (!!this.parent !== !!key.parent) {
```
Comment (devnaumov) [inline webapp/packages/core-sdk/src/Resource/ResourceAlias.ts]: Looks strange, do we need double negation here?

**Review by devnaumov — APPROVED** [review]

File: webapp/packages/core-app/src/CoreSettingsService.ts (hunk 1 above)
Comment (Wroud) [reply webapp/packages/core-app/src/CoreSettingsService.ts:16]: cuz now we can load data page by page, loading 1000 items will take long time and it's not needed in most cases

File: webapp/packages/plugin-navigation-tree/src/NavigationTree/ElementsTree/NavTreeLimitFilter/elementsTreeLimitRenderer.tsx (hunk 2 above)
Comment (Wroud) [reply webapp/packages/plugin-navigation-tree/src/NavigationTree/ElementsTree/NavTreeLimitFilter/elementsTreeLimitRenderer.tsx]: added

File: webapp/packages/core-sdk/src/Resource/ResourceAlias.ts (hunk 3 above)
Comment (Wroud) [reply webapp/packages/core-sdk/src/Resource/ResourceAlias.ts]: you are right, fixed

**Pull request merged**